model/common/src/icon4py/model/common/model_options.py

```diff
@@ -48,13 +49,21 @@ def get_options(program_name: str, **backend_descriptor: Any) -> model_backends.
 
 
 def customize_backend(
-    program_name: str,
-    backend: model_backends.DeviceType | model_backends.BackendDescriptor,
-) -> gtx_typing.Backend:
+    program: gtx_typing.Program | gtx.typing.FieldOperator | None,
+    backend: gtx_typing.Backend
+    | model_backends.DeviceType
+    | model_backends.BackendDescriptor
+    | None,
+) -> gtx_typing.Backend | None:
+    program_name = program.__name__ if program is not None else ""
+    if isinstance(backend, gtx_backend.Backend) or backend is None:
```

Comment:
Doesn't change the result, but maybe swap the two conditions:
```suggestion
    if backend is None or isinstance(backend, gtx_backend.Backend):
```